pkg/build/controller/controller.go

```diff
@@ -284,13 +286,24 @@ func (bc *BuildPodController) HandlePod(pod *kapi.Pod) error {
 	build := obj.(*buildapi.Build)
 
 	nextStatus := build.Status.Phase
+
 	switch pod.Status.Phase {
 	case kapi.PodRunning:
 		// The pod's still running
+		build.Status.Reason = ""
 		nextStatus = buildapi.BuildPhaseRunning
+	case kapi.PodPending:
+		nextStatus = buildapi.BuildPhasePending
```

Comment:
needs to set build.Status.Reason="" first, to ensure the old reason (if any) is cleared, if we change the state to pending and do not have a pushsecret issue.